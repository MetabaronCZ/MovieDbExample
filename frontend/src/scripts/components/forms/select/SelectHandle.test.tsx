import { act } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';
import { fireEvent, getByRole, getByText } from '@testing-library/dom';

import { TestComponentWrapper, waitForComponent } from 'test-utils/component';
import { SelectHandle } from 'components/forms/select/SelectHandle';

describe('components/forms/select/SelectHandle', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <SelectHandle id="ID" onClick={() => null}>
        CONTENT
      </SelectHandle>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const ico = container.querySelector('svg');
    expect(ico).toBeInTheDocument();

    const content = getByText(container, 'CONTENT');
    expect(content).toBeInTheDocument();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('id', 'ID');
    expect(button).toHaveAttribute('title', 'Otevřít');
  });

  it('should reflect closed state', async () => {
    const { container } = await waitForComponent(
      <SelectHandle onClick={() => null}>CONTENT</SelectHandle>,
      { wrapper: TestComponentWrapper },
    );
    const ico = container.querySelector('svg.fa-angle-down');
    expect(ico).toBeInTheDocument();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('title', 'Otevřít');
  });

  it('should reflect opened state', async () => {
    const { container } = await waitForComponent(
      <SelectHandle opened onClick={() => null}>
        CONTENT
      </SelectHandle>,
      { wrapper: TestComponentWrapper },
    );
    const ico = container.querySelector('svg.fa-angle-up');
    expect(ico).toBeInTheDocument();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('title', 'Zavřít');
  });

  it('should callback on click', async () => {
    const onClick = jest.fn();

    const { container } = await waitForComponent(
      <SelectHandle onClick={onClick}>CONTENT</SelectHandle>,
      { wrapper: TestComponentWrapper },
    );
    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    expect(onClick).not.toHaveBeenCalled();

    // click on button
    act(() => {
      fireEvent.click(button);
    });

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should not callback when disabled', async () => {
    const onClick = jest.fn();

    const { container } = await waitForComponent(
      <SelectHandle disabled onClick={onClick}>
        CONTENT
      </SelectHandle>,
      { wrapper: TestComponentWrapper },
    );
    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    expect(onClick).not.toHaveBeenCalled();

    // click on button
    act(() => {
      fireEvent.click(button);
    });

    expect(onClick).not.toHaveBeenCalled();
  });
});
