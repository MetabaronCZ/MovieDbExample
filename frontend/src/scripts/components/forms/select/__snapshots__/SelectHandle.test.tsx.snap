// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/forms/select/SelectHandle should render properly 1`] = `
.c0 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

.c2 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
}

.c1:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

.c3 {
  font-size: 14px;
  font-weight: 400;
  text-align: left;
}

<div>
  <button
    class="c0 c1"
    id="ID"
    title="Otevřít"
    type="button"
  >
    <span
      class="c2 c3"
    >
      CONTENT
    </span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-angle-down fa-lg "
      data-icon="angle-down"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
